File: concrete/src/test/java/pcb/uwu/amount/derived/optics/LuxTest.java
package pcb.uwu.amount.derived.optics;

import org.junit.Test;
import pcb.uwu.amount.derived.fundamental.area.SquareMeters;

import static java.math.MathContext.DECIMAL64;
import static org.junit.Assert.assertEquals;
import static pcb.uwu.core.Magnitude.MILLI;

public class LuxTest {
	private static final Lux lux = new Lux(2000);

	@Test
	public void multipliedByArea() {
		assertEquals(new Lumens(200),
				lux.multipliedBy(new SquareMeters(100, MILLI), DECIMAL64));
	}
}

File: concrete/src/test/java/pcb/uwu/amount/derived/optics/LuxTest.kt
package pcb.uwu.amount.derived.optics

import org.junit.Assert.assertEquals
import org.junit.Test
import pcb.uwu.amount.derived.fundamental.area.SquareMeters
import pcb.uwu.core.Magnitude.MILLI
import java.math.MathContext.DECIMAL64

class LuxTest
{
    @Test
    fun multipliedByArea() =
        assertEquals("200 lm",
                     Lux(2000).multipliedBy(SquareMeters(100, MILLI), DECIMAL64).toString())

}